### DISCUSSION TITLE
Unable to run any version after version 6

### DISCUSSION THREAD

**Discussion by atunguyd** [universam1/iSpindel#586] [General]
``Hi I am building two new iSpindle's for some mates. We sourced the following wemos D1 mini:
https://www.diyelectronics.co.za/store/iot/1971-esp8266-wemos-d1-mini-wifi-dev-board.html

When I flash firmware 5.9.1 in them they work just fine and I can access the config page by pressing the reset button three times and then entering the iSpindle wifi network. However when I try to upgrade to any version from 6.0 and upwards the hotspot is not created. 
Looking on a serial terminal I see the below messages - seems to me like there are different baud rates being used to transmit however I can clearly see what looks like an exception that is occurring.

I do see that after ver 6 the binary size increased to over 512k so I am wondering if there is something different about this particular wemos's memory arrangement? I am not really all that clued up on the wemos devices so I dont know where to go to from here. Any help would be appreciated.

This is the debug I get from 7.2.1:
`rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnnb
bp▒$blrlp▒n0
▒
l
▒
b
n▒n▒
▒
b▒▒nn'l▒l`▒nn
$`nr▒▒▒n
rr▒`p▒n
r▒b
b
n▒n
b▒▒nn'
l`▒nn
$`nr▒▒▒n
rl`r▒▒n
rl`▒
▒bl`▒n▒`rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnnb
bp▒$blrlp▒n
▒
l
▒
b
n▒n▒
▒
b▒▒nn'l▒l`▒nn
$`nr▒▒▒n
rr▒`p▒n0
r▒b
b
n▒n
b▒▒nn'
l`▒nn
$`nr▒▒▒n
rl`r▒▒n
rl`▒
▒bl`▒n▒`rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnn2b
bp▒$blrlp▒n
▒
l
▒
b
n▒n▒
▒
b▒▒nn'l▒l`▒nn
$`nr▒▒▒n
rr▒`p▒n▒
r▒b
b
n▒n
b▒▒nn'
l`▒nn
$`nr▒▒▒n
rl`s▒▒n
rl`▒
▒bl`▒n▒`
        FW 7.2.1
2.2.2-dev(38a443e)
Worker run!
mounting FS... mounted!
ERROR: failed to load json config

ERROR config corrupted
scanning for OW device on pin: 5
No devices found!
scanning for OW device on pin: 12
No devices found!
scanning for OW device on pin: 0
No devices found!
scanning for OW device on pin: 0
No devices found!
ERROR: cannot find a OneWire Temperature Sensor!
Acc Test Connection ERROR!
offsets not available
Boot-Mode: Deep-Sleep Wake
...........
           ERROR no Wifi credentials

going to Config Mode
Acc Test Connection ERROR!

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (28):
epc1=0x4020fdf9 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

>>>stack>>>

ctx: sys
sp: 3ffffd30 end: 3fffffb0 offset: 0190
3ffffec0:  0030ebba 3fff04a0 3fff42cc 40207260
3ffffed0:  3ffe9063 3fffff10 00000000 60000600
3ffffee0:  0030ebba 3fff04a0 3fff1110 4021060e
3ffffef0:  00000000 3fff04a0 00000000 40206b9f
3fffff00:  4022237f 3fff04a0 3fff0954 40206a64
3fffff10:  00000000 00000000 00000000 402071c0
3fffff20:  3fff04a0 3fff04a0 3fff10f4 402072ec
3fffff30:  0030c76d 3fff04a0 3fff0478 40227896
3fffff40:  40275638 3fff10f4 3fff04a0 40228216
3fffff50:  60000600 3fff10f4 3fff04a0 40275645
3fffff60:  4027568a 3fffdab0 00000000 3fffdcb0
3fffff70:  3fff04b0 3fffdab0 00000000 00000004
3fffff80:  40000f49 40000f49 3fffdab0 40000f49
3fffff90:  40000e19 0008fbd1 00000000 00000005
3fffffa0:  00000000 aa55aa55 000000f6 40105631
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3460, room 16
tail 4
chksum 0xcc
load 0x3fff20b8, len 40, room 4
tail 4
chksum 0xc9
csum 0xc9
v0008fbe0
~ld

FW 7.2.1
2.2.2-dev(38a443e)
Worker run!
mounting FS... mounted!
ERROR: failed to load json config

ERROR config corrupted
scanning for OW device on pin: 5
No devices found!
scanning for OW device on pin: 12
No devices found!
scanning for OW device on pin: 0
No devices found!
scanning for OW device on pin: 0
No devices found!
ERROR: cannot find a OneWire Temperature Sensor!
Acc Test Connection ERROR!
offsets not available
Boot-Mode: Exception
...........
           ERROR no Wifi credentials

going to Config Mode
reboot RFCAL
rll▒r$▒n
▒
l▒
b|▒▒▒rblb▒nnlnn2b
bp▒$blrl▒r▒n2nnbp
`

Normal debug from 5.9.1:
`rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnn2b
bp▒$blrlp▒n▒
▒
l
▒
b
n▒nl▒
b▒▒nn'l▒l`▒nn
l`nr▒▒▒n
b
▒`r▒▒n
b
▒`▒
l`▒nb▒rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnnb
bp▒$blrlp▒n
▒
l
▒
b
n▒nl▒
b▒▒nn'l▒l`▒nn
l`nr▒▒▒n
b
▒`r▒▒n
b
▒`▒
l`rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnn2b
bp▒$blrlp▒n▒
▒
l
▒
b
n▒nl▒
b▒▒nn'l▒l`▒nn
l`nr▒▒▒n
b
▒`r▒▒n
b
▒`▒
l`▒nb▒rll▒r$▒n
▒
l▒
b|▒▒▒rb▒
b▒nnlnn2b
bp▒$blrlp▒n
▒
l
▒
b
n▒nl▒
b▒▒nn'l▒l`▒nn
l`nr▒▒▒n
b
▒`r▒▒n
b
▒`▒
l`▒nb▒
      FW 5.9.1
2.2.1(cfd48f3)
Worker run!
Boot-Mode: 6
power-cycle or reset detected, config mode

Double Reset detected
mounting FS... mounted!

ERROR config corrupted
...........
           ERROR no Wifi credentials

going to Config Mode
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
*WM: Scan done
*WM: DUP AP: Desmarais_Home
*WM: DUP AP: Desmarais_Home
*WM: DUP AP: Blackwood_Guest
*WM: DUP AP: Blackwood_Guest
*WM: Adding parameter
*WM: name
*WM: Adding parameter
*WM: sleep
*WM: Adding parameter
*WM: vfact
*WM: Adding parameter
*WM:
*WM: Adding parameter
*WM:
*WM: Adding parameter
*WM: tempscale
*WM: Adding parameter
*WM:
*WM: Adding parameter
*WM:
*WM: Adding parameter
*WM: selAPI
*WM: Adding parameter
*WM: token
*WM: Adding parameter
*WM: server
*WM: Adding parameter
*WM: port
*WM: Adding parameter
*WM: url
*WM: Adding parameter
*WM: db
*WM: Adding parameter
*WM: username
*WM: Adding parameter
*WM: password
*WM: Adding parameter
*WM: job
*WM: Adding parameter
*WM: instance
*WM: Adding parameter
*WM:
*WM: Adding parameter
*WM: POLYN
started Portal
*WM: SET AP
*WM:
*WM: Configuring access point...
*WM: iSpindel
*WM: AP IP address:
*WM: 192.168.4.1
*WM: HTTP server started
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I2C ERROR: 0 con:0
ERROR: OW DISCONNECTED
I
`

**Comment by atunguyd** [comment]
HI Update to this - I was doing this test with a Wemos D1 raw - not connected up to the iSpindle Hardware. My intent was to prepare the wemos device before soldering up the boards. 

Looks like from Version 6 onwards the wemos needs to be in circuit else you get this error. 
Leaving this here incase anyone else hits this "issue"

**Reply by universam1** [reply]
It was always the requirement to connect all hardware before flashing the WeMos

**Reply by atunguyd** [reply]
Thanks - I made the assumption that since I was able to flash and run the code on versions lower than 5.9 that it was the case too on the latest versions. Maybe it would be better to just remove this discussion?
